Std/Data/Nat/Lemmas.lean

```diff
@@ -356,10 +356,10 @@ protected theorem eq_zero_of_add_eq_zero_right (h : n + m = 0) : n = 0 :=
 protected theorem eq_zero_of_add_eq_zero_left (h : n + m = 0) : m = 0 :=
   (Nat.eq_zero_of_add_eq_zero h).2
 
-protected theorem add_eq_zero {n m : Nat} : n + m = 0 ↔ n = 0 ∧ m = 0 :=
-  ⟨eq_zero_of_add_eq_zero, by simp (config := {contextual := true})⟩
+protected theorem add_eq_zero_iff : n + m = 0 ↔ n = 0 ∧ m = 0 :=
+  ⟨Nat.eq_zero_of_add_eq_zero, fun ⟨h₁, h₂⟩ => h₂.symm ▸ h₁⟩
 
-protected theorem add_left_cancel_iff {n m k : Nat} : n + m = n + k ↔ m = k :=
+protected theorem add_left_cancel_iff {n : Nat} : n + m = n + k ↔ m = k :=
```

Comment:
I would actually be in favor of making `n` explicit here but that's debatable. Just voicing an opinion here, this is not a required change.